### DISCUSSION TITLE
Is there a way to run a single @ArchTest through the IDE?

### DISCUSSION THREAD

**Discussion by contivero** [TNG/ArchUnit#1037] [Q&A]
For normal JUnit tests (methods annotated with `@Test`) it's usually possible to right click on them and run them as JUnit test, which means only that test method is run, not all tests in the class. Although I like the succinctness of defining rules as static variables and annotating them with `@ArchTest`, it's inconvenient / wasteful that I cannot run a single rule, but I am rather forced to run all tests in the class. I'm not sure if this is merely an IDE limitation (Eclipse in this case), or if there is something that can be done from ArchUnit's side to support this (in the latter case, I could open an issue about it).

Is there maybe a way I'm missing?

**Comment by hankem** [comment]
This is indeed a current limitation, cf. #641.

You might be more lucky by using [`@ArchTest` annotated methods](https://www.archunit.org/userguide/html/000_Index.html#_writing_tests). If that works and you only need it rarely to debug individual tests, I'd probably create a wrapper on demand:
```java
@ArchTest
void ideExecutableArchTestWrapper(JavaClasses classes) {
    STATIC_FIELD_ARCH_RULE.check(classes);
}
```

**Reply by contivero** [reply]
Thanks for the prompt reply! I see the current options are annotated methods, or commenting out rules that one doesn't want to run (as mentioned in the #641 discussion). I sure hope this is improved somehow in the future, it's not a significant issue but mostly a usability annoyance. Anyway, thanks again and loving the library!
